WcaOnRails/app/assets/javascripts/application.js

```diff
@@ -47,6 +48,64 @@ wca.cancelPendingAjaxAndAjax = function(id, options) {
   return wca._pendingAjaxById[id];
 };
 
+// Adopted from http://stackoverflow.com/a/21778615
+$.fn.scrollToCenter = function(speed) {
+  speed = speed || 200;
+  var el = this;
+  var elOffset = el.offset().top;
+  var windowHeight = $(window).height();
+  var offset = elOffset - Math.max((windowHeight - el.height()) / 2, 0);
+
+  $('html, body').animate({ scrollTop: offset }, speed);
+};
+
+function initMap(competitions, mapElement) {
+  var $map = new google.maps.Map(document.getElementById(mapElement), {
+    zoom: 2,
+    center: {lat: 0, lng: 0},
+    scrollwheel: true
+  });
+
+  var markers = [];
+
+  competitions.forEach(function(c) {
+
+    var contentString = "<a href=" + c.url + ">" + c.name + "</a><br />" + c.marker_date + " - " + c.cityName;
+
+    var infowindow = new google.maps.InfoWindow({
+      content: contentString
+    });
+
+    if (c.is_over) {
+      iconImage = 'http://maps.google.com/mapfiles/ms/icons/blue.png';
+    }
```

Comment:
nitpick: we write our else statements on the same line as the previous closing `}`, so more like

```javascript
if(...) {
  ...
} else {
  ...
}
```